lib/statsd_middleware.rb

```diff
@@ -12,15 +12,33 @@ def call(env)
     status, headers, response = @app.call(env)
     duration = (Time.current - start_time) * 1000.0
 
-    controller = env['action_dispatch.request.path_parameters'][:controller]
-    action = env['action_dispatch.request.path_parameters'][:action]
+    path_parameters = env['action_dispatch.request.path_parameters']
 
-    duration_tags = ["controller:#{controller}", "action:#{action}"]
-    status_tags = duration_tags + ["status:#{status}"]
+    # When ActionDispatch middleware is not processed, as is the case when middleware
```

Comment:
Could probably move this up further in the middleware chain to avoid any refactor required here.

Reply:
But then we'd be measuring duration of throttled requests, i.e. the rack attack processing duration. I think that's supremely unlikely to be a big problem given the scope of problem we expect and how uncritical duration metrics are to our operations, but I don't see a problem with this branch logic.